target/linux/ar71xx/files/arch/mips/ath79/routerboot.c

```diff
@@ -329,6 +329,8 @@ rb_get_board_name(void)
 	int err;
 
 	err = rb_find_hard_cfg_tag(RB_ID_BOARD_NAME, &tag, &tag_len);
+	if (err)
+		err = rb_find_hard_cfg_tag(RB_ID_BOARD_PRODUCT_CODE, &tag, &tag_len);
 	if (err)
 		return NULL;
 
```

Comment:
I added some prints showing the data after each call to `rb_find_hard_cfg_tag`

It seem that RB_ID_BOARD_NAME is present but does not contain valid data.
Hence the Err is not triggered and RB_ID_BOARD_PRODUCT_CODE is never checked.

```
[    0.138864] rb: board name: tag_len: 4  tag: 
[    0.143811] MIPS: machine is Mikrotik RouterBOARD 
[    0.389615] rb: Decompressing with LZO
```

Reply:
Just don't change RB_ID_BOARD_NAME, and remove that change entirely.
Besides, there really is no need to use memcmp to test for a null string 😛